### DISCUSSION TITLE
Will Open3D automatically use cuda to speed up?

### DISCUSSION THREAD

**Discussion by etiwal** [isl-org/Open3D#3606] [Q&A] [answered]
Is Open3D automatically choosing the most performant method for building KDTrees etc.?

I'm using the icp registration module and i can't see any speedup from v.12.0 to v.13.0

**Comment by yxlao** [comment]
The new CUDA-accelerated stuff is in the `open3d::core` and `open3d::t` namespace, where `t` stands for `Tensor`.
Check out https://github.com/intel-isl/Open3D/blob/master/cpp/open3d/t/pipelines/registration/Registration.h

**Reply by etiwal** [reply]
Thanks for your answer! I'm using the Python API. 

Should i first move the target point cloud to the GPU and then apply ICP registration on this? If so how would i move it? Similar to pytorchs point_cloud.to_device()?

**Reply by etiwal** [reply]
Dear @yxlao 

Do you have an update on how to use the ICP registration on GPU through the Python API?

**Reply by ssheorey** [reply, marked as answer]
http://www.open3d.org/docs/latest/tutorial/t_pipelines/t_icp_registration.html#ICP-registration
